homeassistant/components/generic/config_flow.py

```diff
@@ -410,3 +461,41 @@ async def async_step_init(
             ),
             errors=errors,
         )
+
+
+class CameraImagePreview(HomeAssistantView):
+    """Camera view to temporarily serve an image."""
+
+    url = "/api/generic/preview_flow_image/{flow_id}"
+    name = "api:generic:preview_flow_image"
+    requires_auth = False
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialise."""
+        self.hass = hass
+
+    async def get(self, request: web.Request, flow_id) -> web.Response:
+        """Start a GET request."""
+        _LOGGER.debug("processing GET request for flow_id=%s", flow_id)
+        manager: ConfigEntriesFlowManager = self.hass.config_entries.flow
+
+        # @todo - Need to find a more legal way of doing this:
```

Comment:
Ahh.. so there isn't an easy way to get the flow back.

I imagine abusing "async_get" "async_progress" or "async_configure" with a FlowResult return value isn't quite the right way to do this... but maybe it works....

Reply:
Oh I think `async_get` will give you a `FlowResult` which has a `handler` on it which is your `GenericIPCamConfigFlow`?

Reply:
I don't think `handler` helps here.  It's just a string that contains the word `'generic'`;

I cant see a way to get from that back to the configflow object.